src/openms/source/METADATA/ProteinHit.cpp

```diff
@@ -144,6 +144,14 @@ namespace OpenMS
     sequence_.trim();
   }
 
+  // sets the protein sequence
+  void ProteinHit::setSequence(String&& sequence)
+  {
+    sequence_ = std::move(sequence);
+    sequence_.trim();
```

Comment:
Not sure this is a good idea, since you do not know what will happen next.
Also: one function, one purpose. This is two purposes....

If one is, say, reading protein sequences from a file and creates ProtHits from them,
the more efficient interface is the old one, since you keep the original sequence lenght allocated in the reader and just need to reset its value without realloc.

Reply:
What would you suggest? Only add an OPENMS_PRECONDITION that the sequence is trimmed?

I agree that moving could also make things less efficient depending on the context (like in the FASTA reader) but this can be handled in the FASTA reader (e.g., we could remove the move there).

Reply:
I would suggest to simply remove the `trim()`.
The user can decide to trim beforehand if he decides it is reasonable.

Reply:
My guess that this is there for a reason and removing it now will break things. If you volunteer to track that down to parsers etc. we can do ;)

Reply:
I think what would work without the trim ist ph.getSequence().swap or ph.getSequence() = std::move(...)

Reply:
yes, that works, but it feels hacky.
I'd rather volunteer to check (and fix) the occurrences :)

removing the trim would per se not break anything, just use more memory.

Reply:
quickly looking though occurrences of ProteinHit::setSequence:
most of the time, the sequence is set on fresh objects and never touched again.
This means that trim() is a noop.
In a few cases, a proteinhit is used repeadedly (e.g. [in ](https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/FORMAT/HANDLERS/ConsensusXMLHandler.cpp#L454). Also there, the way the code is currently written, trimming does nothing. 
But, if the code were written better (e.g. reusing the allocated ProteinHit, instead of resetting it in every loop by doing `prot_hit_ = ProteinHit();`, then the trimming would actually be a pessimisation..

Please do not trim :)

Reply:
I am just saying it might be there for a reason ;) e.g. parsing external files